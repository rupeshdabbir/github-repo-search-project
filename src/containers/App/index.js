// @flow

import React from 'react';
import { HashRouter as Router, Redirect, Switch } from 'react-router-dom';
import SearchScreen from 'screens/Search';
import DefaultLayout from 'screens/Default';

import 'css/global.css';

export default function App() {
  return (
    <Router>
      <Switch>
        <DefaultLayout exact={true} path="/search" component={SearchScreen} />
        <Redirect to="/search" />
      </Switch>
    </Router>
  );
}
