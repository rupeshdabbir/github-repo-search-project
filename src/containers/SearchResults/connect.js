// @flow

import { connect } from 'react-redux';
import get from 'lodash/fp/get';

export function mapStateToProps(state: Object): Object {
  return {
    userEntities: get('entities.users', state),
    userIds: get('search.result', state),
    totalResults: get('search.totalResults', state),
    pagination: get('search.pagination', state),
    isPending: get('search.isPending', state),
    totalResponse: get('search.totalResponse.items', state),
  };
}

export default connect(mapStateToProps);
