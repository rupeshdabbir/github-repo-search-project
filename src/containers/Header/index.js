// @flow

import React, { Component } from 'react';
import { StyleSheet } from 'aphrodite/no-important';
import Header from 'components/Header';
import Container from 'components/Container';
import connect from './connect';

type Props = {
  onSubmit: Function,
  searchQuery: string,
  searchTerm: string,
  searchForRepos: Function,
};

export class HeaderContainer extends Component {
  static defaultProps = {
    searchTerm: '',
  };

  constructor(props: Props) {
    super(props);
    this.handleSearchRepos(props.searchQuery);
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.searchQuery !== this.props.searchQuery) {
      this.handleSearchRepos(nextProps.searchQuery);
    }
  }

  handleSearchRepos(searchQuery: string): void {
    if (!searchQuery) {
      return;
    }
    this.props.searchForRepos(searchQuery);
  }

  render() {
    const { onSubmit, searchTerm } = this.props;

    return (
      <Container rootStyle={styles.HeaderContainer}>
        <Header onSubmit={onSubmit} searchTerm={searchTerm} />
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  HeaderContainer: {
    backgroundColor: '#f4f4f4',
    borderBottom: '1px solid #aaa',
    boxShadow: '0px 0px 2px 1px rgba(0, 0, 0, .2)',
  },
});

export default connect(HeaderContainer);
