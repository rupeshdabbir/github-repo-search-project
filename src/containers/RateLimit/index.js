import React, { Component } from 'react';
import { css, StyleSheet } from 'aphrodite/no-important';
import 'suitcss-utils-flex/lib/flex.css';
import RateLimit from 'components/RateLimit';
import isUndefined from 'lodash/fp/isUndefined';
import connect from './connect';

type Props = {
  coreLimit: ?number,
  dispatch: Function,
  searchLimit: ?number,
  searchRemaining: ?number,
  searchReset: ?string,
};

export class RateLimitContainer extends Component {
  constructor(props: Props) {
    super(props);
    const { coreLimit, searchLimit, dispatch } = props;

    if (isUndefined(coreLimit) || isUndefined(searchLimit)) {
      dispatch({ type: 'API_RATE_LIMIT_REQUEST' });
    }
  }

  render() {
    const { searchLimit, searchRemaining, searchReset } = this.props;

    return (
      <div className="u-flex u-flexAlignItemsCenter">
        <p className={css(styles.RateLimitContainer_label)}>API requests</p>
        <div className={`${css(styles.RateLimitContainer)} u-flex`}>
          <div className={css(styles.RateLimitContainer_item)}>
            <RateLimit label="Search" remaining={searchRemaining} limit={searchLimit} reset={searchReset} />
          </div>
        </div>
      </div>
    );
  }
}

const styles = StyleSheet.create({
  RateLimitContainer: {
    backgroundColor: '#333',
    color: '#eee',
    padding: '5px 8px',
    borderRadius: 5,
  },

  RateLimitContainer_label: {
    fontSize: 14,
    marginRight: 10,
  },

  RateLimitContainer_item: {
    ':first-child': {
      marginRight: 5,
    },

    ':last-child': {
      marginLeft: 5,
    },
  },
});

export default connect(RateLimitContainer);
