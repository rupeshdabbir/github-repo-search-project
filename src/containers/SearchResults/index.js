// @flow

import React from 'react';
import { css, StyleSheet } from 'aphrodite/no-important';
import SearchTable from 'components/SearchTable';
import Pagination from 'components/Pagination';
import ResultsMessage from 'components/ResultsMessage';
import ErrorBoundary from 'components/ErrorBoundary';
import Loading from 'components/Loading';
import Container from 'components/Container';
import connect from './connect';

type Props = {
  isPending: boolean,
  pagination: ?Object,
  searchTerm: string,
  totalResults: number,
  userIds: Array<number>,
};

export function SearchResultsContainer(props: Props) {
  const { isPending, pagination, searchTerm, totalResults, userIds, totalResponse } = props;

  console.log('props', props);

  if (!searchTerm) {
    return null;
  }

  if (isPending) {
    return (
      <div className={css(styles.SearchResultsContainer_wrapLoading)}>
        <Loading />
      </div>
    );
  }

  return (
    <Container>
      <div className={css(styles.SearchResultsContainer_item, styles.SearchResultsContainer_wrapResultsMessage)}>
        <ErrorBoundary>
          <ResultsMessage searchTerm={searchTerm} resultsTotal={totalResults} pageTotal={userIds.length} />
        </ErrorBoundary>
      </div>
      <div className={css(styles.SearchResultsContainer_item)}>
        <ErrorBoundary>
          {/* <SearchResults searchTerm={searchTerm} ids={userIds} entities={userEntities} total={totalResults} /> */}
          <SearchTable totalResponse={totalResponse} />
        </ErrorBoundary>
      </div>
      <div className={css(styles.SearchResultsContainer_item)}>
        <ErrorBoundary>
          <Pagination {...pagination} />
        </ErrorBoundary>
      </div>
    </Container>
  );
}

SearchResultsContainer.defaultProps = {
  pagination: {},
  searchTerm: '',
};

const styles = StyleSheet.create({
  SearchResultsContainer_wrapResultsMessage: {
    marginBottom: 30,
  },

  SearchResultsContainer_wrapLoading: {
    marginTop: 25,
  },

  SearchResultsContainer_item: {
    marginTop: 15,
  },
});

export default connect(SearchResultsContainer);
