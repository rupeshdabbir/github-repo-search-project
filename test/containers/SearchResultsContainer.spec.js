import React from 'react';
import renderer from 'react-test-renderer';
import {StyleSheetTestUtils} from 'aphrodite';

import {SearchResultsContainer} from 'containers/SearchResults';

describe('Component: SearchResultsContainer', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  describe('when the searchTerm is falsey', () => {
    it('should not render the pagination or results', () => {
      const component = renderer.create(
        <SearchResultsContainer
          isPending={true}
          searchTerm={''}
          userEntities={{}}
          userIds={[]}
          totalResults={0}
        />
      );
      expect(component).toMatchSnapshot();
    });
  });

});
