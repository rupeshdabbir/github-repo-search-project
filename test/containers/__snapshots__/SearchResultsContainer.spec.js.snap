// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SearchResultsContainer when the searchTerm is falsey should not render the pagination or results 1`] = `null`;
